test/test.js

```diff
@@ -52,284 +58,224 @@ describe('YeldaChat', () => {
       }
-
-      yeldaChat.init(mockData)
+      yeldaChat.unLoadChat()
+      await yeldaChat.init(mockData)
       expect(yeldaChat.webChatContainer).not.to.be.null
     })
   })
 
   describe('yeldaChat.setupChat', () => {
-
     describe('yeldaChat.setupChat initial checks', () => {
-      it('should return null if assistantId is missing', () => {
+      it('should webChatContainer is null if assistantId is missing', async () => {
         // Reset the DOM
         yeldaChat.unLoadChat()
         const mockData = {
           'assistantSlug': 'testClient',
           'chatPath': 'chat'
         }
 
-        const response = yeldaChat.setupChat(mockData)
-
-        assert.typeOf(response, 'null', 'setupChat function returns null')
+        await yeldaChat.setupChat(mockData)
         expect(yeldaChat.webChatContainer).to.be.null
       })
-      it('should return null if assistantSlug is missing', () => {
+
+      it('should webChatContainer is null if assistantSlug is missing', async () => {
         // Reset the DOM
         yeldaChat.unLoadChat()
         const mockData = {
           'assistantId': 'testClient',
           'chatPath': 'chat'
         }
 
-        const response = yeldaChat.setupChat(mockData)
-
-        assert.typeOf(response, 'null', 'setupChat function returns null')
+        await yeldaChat.setupChat(mockData)
         expect(yeldaChat.webChatContainer).to.be.null
       })
     })
 
-  describe('yeldaChat.createContainer', () => {
-    it('should create webChatContainer dom element', () => {
+    describe('yeldaChat.createContainer', async() => {
       // Reset the DOM
       yeldaChat.unLoadChat()
-      const mockData = {
-        'assistantSlug': 'testClient',
-        'assistantId': '12345678',
-        'chatPath': 'chat'
-      }
-
-      yeldaChat.setupChat(mockData)
-      expect(yeldaChat.webChatContainer).not.to.be.null
-    })
-
-    it('should define parentContainer', () => {
-      expect(yeldaChat.parentContainer).not.to.be.null
-    })
-
-    it('expect webChatContainer exists in document', () => {
-      document.querySelector('#yelda_container').should.exist
-    })
-
-    it('webChatContainer should have attribute id', () => {
-      expect(yeldaChat.webChatContainer).to.have.attribute('id', 'yelda_container')
-    })
-
-    it('webChatContainer should have attribute class', () => {
-      expect(yeldaChat.webChatContainer).to.have.attribute('class', 'yelda_container')
-    })
-
-    it('expect parentContainer contain webChatContainer', () => {
-      expect(yeldaChat.parentContainer).to.contain(yeldaChat.webChatContainer)
-    })
-  })
-
-  describe('yeldaChat.addAssistantImage', () => {
-    it('should create assistantImage dom element', () => {
-      yeldaChat.should.have.property('assistantImage')
-    })
-
-    it('expect yelda_assistant_img exists in document', () => {
-      document.querySelector('#yelda_assistant_img').should.exist
-    })
+      await yeldaChat.setupChat(validMockData)
```

Comment:
For each `describe` block that you want to have async operation before the tests, you can use the `before()` method so the tests will stay in the same order in the output.
```js
before(async () => {
  yeldaChat.unLoadChat()
  await yeldaChat.setupChat(validMockData)
})
```